waspc/cli/src/Wasp/Cli/Command/CreateNewProject.hs

```diff
@@ -18,27 +18,42 @@ import Wasp.Cli.Command (Command, CommandError (..))
 import qualified Wasp.Cli.Command.Common as Command.Common
 import qualified Wasp.Cli.Common as Common
 import qualified Wasp.Data
-import Wasp.Util (indent)
+import Wasp.Util (indent, kebabToCamelCase)
 import qualified Wasp.Util.Terminal as Term
 
-newtype ProjectName = ProjectName {_projectName :: String}
+data ProjectInfo = ProjectInfo
+  { _projectName :: String,
+    _appName :: String
+  }
+
+-- Takes a project name String
+-- Returns either the ProjectInfo type that contains both the Project name
+-- and the App name (which might be the same), or an error describing why the name is invalid
+parseProjectNameIntoProjectInfo :: String -> Either String ProjectInfo
```

Comment:
I suggest changing the function's name to `parseProjectInfo` or `getProjectInfo`. Strictly speaking, it isn't correct to say we're parsing `ProjectName` into `ProjectInfo` because the argument string is not a project name until it's successfully parsed and validated (which is what this function does). It's just a String at that point, parsing is what turns it into a `ProjectName`.

Reply:
Got it, thanks. I went with `parseProjectInfo`

Reply:
Ah uff it was me who just in the previous iteration suggested the name `parseProjectNameIntoProjectInfo` :D, sorry @pt2121 !
@sodic I am not sure if semantics you are following here are widely adopted / clear, I personally don't read it that way, I read it more like "I got project name and I am parsing it into project info". That doesn't mean project name is necessarily valid, since functions returns Either obviously there can be an error in parsing, and error in parsing can be cause only by input. Parsing more often than not assumes there could be an error in the process, so that is kind of implicit.
And you are not really parsing ProjectInfo. You are parsing project name. And the result of it is project info. When you look at parse as a verb in English language, when you say `parse X`, you are going through X in order to produce something out of it. So in that fashion, `parseProjectInfo` would mean we receive ProjectInfo and then parse that into something. I think in computer science we do often use it the opposite way though hm. But that is kind of ok because if you have `parseProgram :: String -> Program` that is because both that String represents program, and Program represents program. So you are indeed parsing program in String format into program in Program format.
But here we don't actually have ProjectInfo in the input. We have project name, in the form of string. And the result is ProjectInfo which goes through additional step of deriving app name from the project name.
Whooops this became lengthy! Let's leave it at `parseProjectInfo`, no problem, I don't want to further bother @pt2121 with this :D, but I thought this is interesting discussion.

Reply:
Ah, I like how you guys thoroughly thought about this. In the mean time, hope the code comment helps a bit in term of maintenance. :smiley_cat: